File: android/src/main/kotlin/com/wortise/ads/flutter/extensions/RequestParameters.kt
package com.wortise.ads.flutter.extensions

import com.wortise.ads.CollapsiblePosition
import com.wortise.ads.RequestParameters
import io.flutter.plugin.common.MethodCall

fun getRequestParameters(args: Map<*, *>?): RequestParameters {
    val params = args?.get("requestParameters") as? Map<*, *>

    val agent = params?.get("agent") as? String

    val collapsible = CollapsiblePosition.fromValue(
        params?.get("collapsible") as? String
    )

    return RequestParameters(
        agent       = agent,
        collapsible = collapsible
    )
}

fun MethodCall.getRequestParameters(): RequestParameters =
    getRequestParameters(arguments as? Map<*, *>)

File: lib/src/request_parameters.dart
enum CollapsiblePosition {
  bottom,
  top,
}

class RequestParameters {

  final String? agent;

  final CollapsiblePosition? collapsible;

  const RequestParameters({
    this.agent,
    this.collapsible,
  });


  Map<String, dynamic> get toMap => {
    "agent": agent,
    "collapsible": collapsible?.name,
  };
}
